packages/js-commons/src/local-files.ts

```diff
@@ -0,0 +1,262 @@
+import { join } from "node:path";
+import { existsSync, mkdirSync } from "node:fs";
+import { readFile, writeFile } from "node:fs/promises";
+import type {
+  YamlApplicationTransporter,
+  LocalApplicationTransporter,
+} from "./transporter";
+import {
+  ArtifactFilesName,
+  HiddenManifestFile,
+  AppManifestDataFile,
+  ArtifactYamlFolderNames,
+} from "./constants";
+import type {
+  ThemeDTO,
+  AssetDTO,
+  ScreenDTO,
+  WidgetDTO,
+  ScriptDTO,
+  LanguageDTO,
+  ApplicationDTO,
+  EnvironmentDTO,
+  ApplicationMetaDTO,
+  EnsembleDocument,
+} from "./dto";
+import { EnsembleDocumentType } from "./enums";
+
+export const getLocalApplicationTransporter = (
+  hiddenEnsembleFolder: string,
+): LocalApplicationTransporter => ({
+  get: async (appId: string): Promise<ApplicationDTO | null> => {
+    const userAppFolder = join(hiddenEnsembleFolder, appId);
```

Comment:
The user app folder should not be inside the hidden ensemble folder. The hidden ensemble folder keeps track of the app paths on the local FS.

Reply:
The structure is like that :
1- A hidden global folder that contains a file that holds metadata of all apps and yaml folder paths on user local disk.
2- Each app is downloaded in a separate folder with the app name. Each such app folder contains a hidden folder with name `.ensemble` and a yaml folder for each type of artifact `Screens`, `Widgets`,  `Theme`, `Scripts` and `Translations`.
3- Each Yaml folder contains yaml files with the name of artifact and a hidden `.manifest.json` file to keep track of names of files with the artifact id.

This is what we discussed and this is what we agreed on. Let me know if I missed anything.